docs/GettingStarted.md

```diff
@@ -807,8 +807,57 @@ function display(type, value) {
     container.className.replace(RegExp('display-' + type + '-[a-z]+ ?'), '');
   event && event.preventDefault();
 }
-var isMac = navigator.platform === 'MacIntel';
-var isWindows = navigator.platform === 'Win32';
-display('os', isMac ? 'mac' : (isWindows ? 'windows' : 'linux'));
-display('platform', isMac ? 'ios' : 'android');
+
+// If we are coming to the page with a hash in it (i.e. from a search, for example), try to get
+// us as close as possible to the correct platform and dev os using the hashtag and block walk up.
+var foundHash = false;
+if (window.location.hash !== '' && window.location.hash !== 'content') { // content is default
+  var hashLinks = document.querySelectorAll('a.hash-link');
+  for (var i = 0; i < hashLinks.length && !foundHash; ++i) {
+    if (hashLinks[i].hash === window.location.hash) {
+      var parent = hashLinks[i].parentElement;
+      while (parent !== null) {
```

Comment:
I'm 95% sure that it will either do an infinite loop or throw an error. parentElement will be undefined and not null if it doesn't exist. Just doing if (parent) { would be better